dom.bs

```diff
@@ -5751,9 +5757,21 @@ is an object or one of its <a>shadow-including ancestors</a>.
 <var>B</var>, repeat these steps until they return an object:</p>
 
 <ol>
- <li><p>If <var>A</var>'s <a for=tree>root</a> is not a <a for=/>shadow root</a>, or <var>A</var>'s
- <a for=tree>root</a> is a <a>shadow-including inclusive ancestor</a> of <var>B</var>, then return
- <var>A</var>.
+ <li><p>If <var>A</var> is a {{Window}} object, then set <var>A</var> to <var>A</var>'s
+ <a>associated <code>Document</code></a>.
```

Comment:
Hmm, retargeting Window gives Document? Shouldn't this just return Window

Reply:
Hmm yeah, I guess we just want it to behave as if you passed document.

Reply:
I guess I should just remove this step altogether then?

Reply:
Because window is not an instance of a Node, it falls into the line 5767 (A is not a node)?
Sounds good  - but wondering what's the reasoning behind adding this line.

Reply:
I added this because of https://github.com/whatwg/dom/issues/580#issuecomment-369813057, but I didn't really think it through.

Reply:
I think removing this step is not enough. 
e.g. If A is a node in a shadow tree and B is a Window, we should loop until A reaches the outer most shadow host.

The following should work, I think.

1. If one of the following is true

   - A is not a node
   - A’s root is not a shadow root
   - B is a node and A’s root is a shadow-including inclusive ancestor of B

   then return A.

2. Set A to A’s root’s host.

Reply:
Thanks, that does look good.